### PR TITLE
Shim fixes

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

**Comment by dlespiau** [comment]
lgtm

**Comment by dlespiau** [comment]
It seems like some of these should be ported to 3.0 as well? The EINTR one at least.

**Comment by amshinde** [comment]
Yes. Will patch these in 3.0 as well

File: shim/shim.c [hunk 1]
```diff
@@ -434,10 +434,7 @@ handle_proxy_output(struct cc_shim *shim)
 
 	buf = read_IO_message(shim, &seq, &stream_len);
 	if ((! buf) || (stream_len <= 0) || (stream_len > HYPERSTART_MAX_RECV_BYTES)) {
-		/*TODO: is exiting here more appropriate, since this denotes
-		 * error communicating with proxy or proxy has exited
-		 */
-		goto out;
+		exit(EXIT_FAILURE);
```
Comment (jodh-intel) [inline shim/shim.c]: A `shim_error()` call could be useful to debug this code path I think?

**Comment by chavafg** [comment]
qa-failed

**Comment by chavafg** [comment]
qa-passed

File: shim/shim.c (hunk 1 above)
Comment (amshinde) [inline shim/shim.c]: Added an error log message here.

**Comment by chavafg** [comment]
qa-passed

**Comment by jcvenegas** [comment]
lgtm

**Pull request merged**